cabal-install/Distribution/Client/Sandbox.hs

```diff
@@ -368,6 +374,47 @@ sandboxDelete verbosity _sandboxFlags globalFlags = do
       notice verbosity $ "Deleting the sandbox located at " ++ sandboxDir
       removeDirectoryRecursive sandboxDir
 
+      absSandboxDir <- makeAbsolute sandboxDir
+
+      let
+        pathInsideSandbox = isPrefixOf absSandboxDir
+
+      let
+        -- Warn the user if CABAL_SANDBOX_PACKAGE_PATH points inside the
+        -- deleted sandbox.
+        checkSandboxPackagePath path = do
```

Comment:
`checkSandboxPackagePath`, `checkGHCPackagePath` and `checkGHCJSPackagePath` look quite similar. Can we merge them into a single function? Also, the `liftM (maybe [] splitSearchPath) ...` bit looks like it could be part of that function as well.